src/NHibernate/Dialect/MsSql2000Dialect.cs

```diff
@@ -66,35 +66,412 @@ protected virtual void RegisterDefaultProperties()
 
 		protected virtual void RegisterKeywords()
 		{
-			RegisterKeyword("top");
-			RegisterKeyword("int");
-			RegisterKeyword("integer"); // a commonly-used alias for 'int'
-			RegisterKeyword("tinyint");
-			RegisterKeyword("smallint");
+			RegisterKeyword("absolute");
```

Comment:
We could optimize this only by providing this array to the `HashSet<>` constructor